### PR TITLE
Feature/xivy 4995 support rest deployments with encrypted maven passwords

### PR DESCRIPTION
**Pull request by ivy-lli**

### CODE REVIEW

File: src/main/java/ch/ivyteam/ivy/maven/engine/deploy/http/HttpDeployer.java [hunk 1]
```diff
@@ -111,14 +115,21 @@ private HttpEntity getRequestData(File resolvedOptionsFile)
...
   {
     String username = "admin";
     String password = "admin";
     if (server != null)
     {
       username = server.getUsername();
-      password = server.getPassword();
+      try
+      {
+        password = secDispatcher.decrypt(server.getPassword());
```
Comment (ivy-rew) [inline src/main/java/ch/ivyteam/ivy/maven/engine/deploy/http/HttpDeployer.java:127]: do plain passwords still work with this API?
would be happy to see a simple test using this feature ... if not easily possible here: then in docker or a third-party repo (docker-samples)?

**Review by ivy-rew — COMMENTED** [review]
be aware that there is an lts8 tag ... very likely you can provide this feature in LTS compatible manor for master+release/8.0

File: src/main/java/ch/ivyteam/ivy/maven/engine/deploy/http/HttpDeployer.java (hunk 1 above)
Comment (ivy-lli) [reply src/main/java/ch/ivyteam/ivy/maven/engine/deploy/http/HttpDeployer.java:127]: Yes I've tried it and it worked with plaintext passwords too. Maven encrypted passwords starts and ends with {}.
Even something like this works:
`<password>This is my encrpyed password. {Wdc6XFJMDk4H8L3PspU513LuqxP3QhLcL/8LhAilBaQ=}</password>`

Comment (ivy-lli) [reply src/main/java/ch/ivyteam/ivy/maven/engine/deploy/http/HttpDeployer.java:127]: But yes as this feature is fully untested we should definitely write some tests somewhere.

**Review by ivy-rew — APPROVED** [review]
absolutely love to see our support being that damn fast and responsible to customer feedback.

**Pull request merged**